### PR TITLE
Added configuration files for BROTHERHOBBYF405V2, BROTHERHOBBYF405V3, and BROTHERHOBBYF722V3 flight controllers

### PR DESCRIPTION
**Pull request by BrotherHobby**
### Description
Added configuration files for BROTHERHOBBYF405V2, BROTHERHOBBYF405V3, and BROTHERHOBBYF722V3 flight controllers. This includes:

- Implementation of the configuration files in `config.h` for each flight controller.
- Updated necessary configuration settings to integrate the new flight controllers.

### Changes
- 3 files changed
- 368 insertions
- Created new files `configs/BROTHERHOBBYF405V2/config.h`, `configs/BROTHERHOBBYF405V3/config.h`, and `configs/BROTHERHOBBYF722V3/config.h`.

### Testing
- Verified the functionality of the new configuration files with various test cases.
- Ensured compatibility with existing system components.

Please review the changes and let me know if any adjustments are needed. Thank you!

### CODE REVIEW

File: configs/BROTHERHOBBYF405V2/config.h [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define FC_TARGET_MCU     STM32F405
+#define BOARD_NAME        BHF405V2
+#define MANUFACTURER_ID   Brot
```
Comment (nerdCopter) [inline configs/BROTHERHOBBYF405V2/config.h:26]: manufacturer id need to be capitals.
also need to add Brother Hobby to `./Manufacturers.md`

File: configs/BROTHERHOBBYF405V2/config.h [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+#define DASHBOARD_I2C_INSTANCE              (I2CDEV_2)
+// #define SERIALRX_PROVIDER                   CRSF
+#define DEFAULT_BLACKBOX_DEVICE             BLACKBOX_DEVICE_FLASH
+#define DEFAULT_DSHOT_BURST                 DSHOT_DMAR_ON
+#define DEFAULT_CURRENT_METER_SOURCE        CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE        VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SCALE         386
+#define BEEPER_INVERTED
+#define FLASH_SPI_INSTANCE                  SPI2
+#define MAX7456_SPI_INSTANCE                SPI2
```
Comment (haslinghuis) [inline configs/BROTHERHOBBYF405V2/config.h:109]: Sharing SPI is not recommended.

File: configs/BROTHERHOBBYF405V2/config.h [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+#define FC_TARGET_MCU     STM32F405
+#define BOARD_NAME        BHF405V2
+#define MANUFACTURER_ID   Brot
+
+#define USE_ACC
+#define USE_ACC_SPI_ICM42688P
+#define USE_GYRO
+#define USE_GYRO_SPI_ICM42688P
+#define USE_BARO
+#define USE_BARO_SPL06
```
Comment (haslinghuis) [inline configs/BROTHERHOBBYF405V2/config.h:33]: SPL06 is handled by DPS310 driver.
```suggestion
```

File: configs/BROTHERHOBBYF405V3/config.h [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+#define FC_TARGET_MCU     STM32F405
+#define BOARD_NAME        BHF405V3
+#define MANUFACTURER_ID   Brot
+
+#define USE_ACC
+#define USE_ACC_SPI_ICM42688P
+#define USE_GYRO
+#define USE_GYRO_SPI_ICM42688P
+#define USE_BARO
+#define USE_BARO_SPL06
```
Comment (haslinghuis) [inline configs/BROTHERHOBBYF405V3/config.h:33]:
```suggestion
```

File: configs/BROTHERHOBBYF722V3/config.h [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+#define FC_TARGET_MCU     STM32F7X2
+#define BOARD_NAME        BHF722V3
+#define MANUFACTURER_ID   Brot
+
+#define USE_ACC
+#define USE_ACC_SPI_ICM42688P
+#define USE_GYRO
+#define USE_GYRO_SPI_ICM42688P
+#define USE_BARO
+#define USE_BARO_SPL06
```
Comment (haslinghuis) [inline configs/BROTHERHOBBYF722V3/config.h:33]:
```suggestion
```

File: configs/BROTHERHOBBYF722V3/config.h [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+#define BARO_I2C_INSTANCE                   (I2CDEV_1)
+// #define SERIALRX_PROVIDER                   CRSF
+#define DEFAULT_BLACKBOX_DEVICE             BLACKBOX_DEVICE_FLASH
+#define DEFAULT_CURRENT_METER_SOURCE        CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE        VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SCALE         386
+#define BEEPER_INVERTED
+#define FLASH_SPI_INSTANCE                  SPI2
+#define MAX7456_SPI_INSTANCE                SPI2   
+#define USE_SPI_GYRO
```
Comment (haslinghuis) [inline configs/BROTHERHOBBYF722V3/config.h:108]: Already defined in target.h

```suggestion
```

File: configs/BROTHERHOBBYF405V3/config.h [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+        TIMER_PIN_MAP(9,  PA3, 2,  0)\
+        TIMER_PIN_MAP(10, PB3, 1,  0)\
+        TIMER_PIN_MAP(11, PC9, 2,  0)
+
+#define SPI2_TX_DMA_OPT     0
+#define ADC1_DMA_OPT        0
+
+#define MAG_I2C_INSTANCE                    (I2CDEV_2)
+#define BARO_I2C_INSTANCE                   (I2CDEV_2)
+#define DASHBOARD_I2C_INSTANCE              (I2CDEV_2)
```
Comment (haslinghuis) [inline configs/BROTHERHOBBYF405V3/config.h:103]: No need to include DASHBOARD

```suggestion
```

File: configs/BROTHERHOBBYF405V3/config.h [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+#define CAMERA_CONTROL_PIN      PB3
+#define ADC_VBAT_PIN            PC0
+#define ADC_RSSI_PIN            PC2
+#define ADC_CURR_PIN            PC1
+#define FLASH_CS_PIN            PA15
+#define MAX7456_SPI_CS_PIN      PB12
+#define GYRO_1_EXTI_PIN         PC4
+#define GYRO_1_CS_PIN           PA4
+
+
```
Comment (haslinghuis) [inline configs/BROTHERHOBBYF405V3/config.h:83]: Excessive white space

```suggestion

```

File: configs/BROTHERHOBBYF405V3/config.h [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+#define DEFAULT_CURRENT_METER_SCALE         386
+#define BEEPER_INVERTED
+#define FLASH_SPI_INSTANCE                  SPI2
+#define MAX7456_SPI_INSTANCE                SPI2   
+#define USE_SPI_GYRO
+#define GYRO_1_SPI_INSTANCE                 SPI1
+#define GYRO_1_ALIGN                        CW90_DEG
+#define SYSTEM_HSE_MHZ                      8       
+
+
```
Comment (haslinghuis) [inline configs/BROTHERHOBBYF405V3/config.h:118]: Excessive white space

```suggestion

```

File: configs/BROTHERHOBBYF405V3/config.h [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+#define SERIALRX_UART                       SERIAL_PORT_USART2
+#define MSP_DISPLAYPORT_UART                SERIAL_PORT_UART5
+
+
+
+
+
```
Comment (haslinghuis) [inline configs/BROTHERHOBBYF405V3/config.h:125]: Excessive white space

```suggestion

```

File: configs/BROTHERHOBBYF405V3/config.h [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+// #define SERIALRX_PROVIDER                   CRSF
+#define DEFAULT_BLACKBOX_DEVICE             BLACKBOX_DEVICE_FLASH
+#define DEFAULT_DSHOT_BURST                 DSHOT_DMAR_ON
+#define DEFAULT_CURRENT_METER_SOURCE        CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE        VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SCALE         386
+#define BEEPER_INVERTED
+#define FLASH_SPI_INSTANCE                  SPI2
+#define MAX7456_SPI_INSTANCE                SPI2   
+#define USE_SPI_GYRO
```
Comment (haslinghuis) [inline configs/BROTHERHOBBYF405V3/config.h:113]: Already defined in target.h

```suggestion
```

File: configs/BROTHERHOBBYF405V2/config.h [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+// #define SERIALRX_PROVIDER                   CRSF
+#define DEFAULT_BLACKBOX_DEVICE             BLACKBOX_DEVICE_FLASH
+#define DEFAULT_DSHOT_BURST                 DSHOT_DMAR_ON
+#define DEFAULT_CURRENT_METER_SOURCE        CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE        VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SCALE         386
+#define BEEPER_INVERTED
+#define FLASH_SPI_INSTANCE                  SPI2
+#define MAX7456_SPI_INSTANCE                SPI2   
+#define USE_SPI_GYRO
```
Comment (haslinghuis) [inline configs/BROTHERHOBBYF405V2/config.h:110]: Already defined in target.h

```suggestion
```

**Review by haslinghuis — CHANGES_REQUESTED** [review]
Sharing SPI is not recommended.

File: configs/BROTHERHOBBYF722V3/config.h [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+#define GYRO_1_SPI_INSTANCE                 SPI1
+#define GYRO_1_ALIGN                        CW90_DEG
+#define SYSTEM_HSE_MHZ                      8       
+#define DEFAULT_DSHOT_BURST                 DSHOT_DMAR_ON
+
+
+#define SERIALRX_UART                       SERIAL_PORT_USART2
+#define MSP_DISPLAYPORT_UART                SERIAL_PORT_UART4
+
```
Comment (SteveCEvans) [inline configs/BROTHERHOBBYF722V3/config.h:117]: As @haslinghuis has pointed out a few times, there are a lot of excessive blank lines.

**Comment by ot0tot** [comment]
No DMA assignment is possible for Motor 4 on the F405 targets, which means the FC must use bitbanged DShot and the gyro on SPI 1 cannot use DMA. This seems like a major oversight and will result in poor performance.  IMO these targets need to be redesigned and should not be approved as-is.

File: configs/BROTHERHOBBYF405V2/config.h (hunk 2 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF405V2/config.h:109]: This is determined according to the actual pin distribution.

File: configs/BROTHERHOBBYF722V3/config.h [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+#define CAMERA_CONTROL_PIN   PB7
+#define ADC_VBAT_PIN         PC2
+#define ADC_RSSI_PIN         PC0
+#define ADC_CURR_PIN         PC1
+#define FLASH_CS_PIN         PB6
+#define MAX7456_SPI_CS_PIN   PB12
+#define GYRO_1_EXTI_PIN      PC4
+#define GYRO_1_CS_PIN        PA4
+
+#define TIMER_PIN_MAPPING \
```
[the remark this answers is not in the document]
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF722V3/config.h:82]: Yes, the timer allocation has been tested.

File: configs/BROTHERHOBBYF405V2/config.h (hunk 1 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF405V2/config.h]: The modification has been completed

File: configs/BROTHERHOBBYF405V2/config.h (hunk 3 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF405V2/config.h]: The modification has been completed

File: configs/BROTHERHOBBYF405V3/config.h (hunk 4 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF405V3/config.h]: The modification has been completed

File: configs/BROTHERHOBBYF722V3/config.h (hunk 5 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF722V3/config.h]: The modification has been completed

File: configs/BROTHERHOBBYF722V3/config.h (hunk 6 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF722V3/config.h]: The modification has been completed

File: configs/BROTHERHOBBYF405V3/config.h (hunk 8 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF405V3/config.h]: The modification has been completed

File: configs/BROTHERHOBBYF405V3/config.h (hunk 9 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF405V3/config.h]: The modification has been completed

File: configs/BROTHERHOBBYF405V3/config.h (hunk 10 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF405V3/config.h]: The modification has been completed

File: configs/BROTHERHOBBYF405V3/config.h (hunk 11 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF405V3/config.h]: The modification has been completed

File: configs/BROTHERHOBBYF405V2/config.h (hunk 12 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF405V2/config.h]: The modification has been completed

File: configs/BROTHERHOBBYF722V3/config.h (hunk 13 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF722V3/config.h]: The modification has been completed

File: configs/BROTHERHOBBYF405V2/config.h (hunk 1 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF405V2/config.h]: Please review again. Thank you!

File: configs/BROTHERHOBBYF405V2/config.h (hunk 2 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF405V2/config.h:106]: Please review again. Thank you!

File: configs/BROTHERHOBBYF405V2/config.h (hunk 3 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF405V2/config.h]: Please review again. Thank you!

File: configs/BROTHERHOBBYF405V3/config.h (hunk 4 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF405V3/config.h]: Please review again. Thank you!

File: configs/BROTHERHOBBYF722V3/config.h (hunk 5 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF722V3/config.h]: Please review again. Thank you!

File: configs/BROTHERHOBBYF722V3/config.h (hunk 6 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF722V3/config.h]: Please review again. Thank you!

File: configs/BROTHERHOBBYF405V3/config.h (hunk 7 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF405V3/config.h]: The modification has been completed，Please review again. Thank you!

**Review by haslinghuis — APPROVED** [review]

File: configs/BROTHERHOBBYF722V3/config.h [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+#define DEFAULT_CURRENT_METER_SCALE         386
+#define BEEPER_INVERTED
+#define FLASH_SPI_INSTANCE                  SPI2
+#define MAX7456_SPI_INSTANCE                SPI2   
+#define GYRO_1_SPI_INSTANCE                 SPI1
+#define GYRO_1_ALIGN                        CW90_DEG
+#define SYSTEM_HSE_MHZ                      8       
+#define DEFAULT_DSHOT_BURST                 DSHOT_DMAR_ON
+#define SERIALRX_UART                       SERIAL_PORT_USART2
+#define MSP_DISPLAYPORT_UART                SERIAL_PORT_UART4
```
Comment (SteveCEvans) [inline configs/BROTHERHOBBYF722V3/config.h:112]: Schematic shows this to be UART 1.

**Review by SteveCEvans — CHANGES_REQUESTED** [review]

**Review by nerdCopter — APPROVED** [review]

File: configs/BROTHERHOBBYF722V3/config.h (hunk 15 above)
Comment (BrotherHobby) [reply configs/BROTHERHOBBYF722V3/config.h:112]: On the schematic, UART1 is connected to the sky end, and UART2 and UART4 are receivers and electric tuners, respectively.

Comment (haslinghuis) [reply configs/BROTHERHOBBYF722V3/config.h:112]: @BrotherHobby Are there wire diagrams available for ports used - to verify

Comment (haslinghuis) [reply configs/BROTHERHOBBYF722V3/config.h:112]: @BrotherHobby If sky end means [digital] FPV output this needs to be UART1

What is electric tuners?

Comment (BrotherHobby) [reply configs/BROTHERHOBBYF722V3/config.h:112]: > @BrotherHobby Are there wire diagrams available for ports used - to verify

Yes, it has been sent

Comment (haslinghuis) [inline configs/BROTHERHOBBYF722V3/config.h:112]:
```suggestion
#define MSP_DISPLAYPORT_UART                SERIAL_PORT_USART1

```

**Review by haslinghuis — CHANGES_REQUESTED** [review]

**Review by haslinghuis — APPROVED** [review]

**Comment by haslinghuis** [comment]
Please address https://github.com/betaflight/config/pull/454#issuecomment-2155181412

**Comment by BrotherHobby** [comment]
> No DMA assignment is possible for Motor 4 on the F405 targets, which means the FC must use bitbanged DShot and the gyro on SPI 1 cannot use DMA. This seems like a major oversight and will result in poor performance. IMO these targets need to be redesigned and should not be approved as-is.

How should this be distributed?

**Comment by BrotherHobby** [comment]
> > No DMA assignment is possible for Motor 4 on the F405 targets, which means the FC must use bitbanged DShot and the gyro on SPI 1 cannot use DMA. This seems like a major oversight and will result in poor performance. IMO these targets need to be redesigned and should not be approved as-is.
> 
> How should this be distributed?

It has been modified, and you need to check whether it matches or not.

**Comment by BrotherHobby** [comment]
> > > > No DMA assignment is possible for Motor 4 on the F405 targets, which means the FC must use bitbanged DShot and the gyro on SPI 1 cannot use DMA. This seems like a major oversight and will result in poor performance. IMO these targets need to be redesigned and should not be approved as-is.
> > > 
> > > 
> > > How should this be distributed?
> > 
> > 
> > It has been modified, and you need to check whether it matches or not.
> 
> Your change is invalid. Re-read my comment.
> 
> You are using pin B09 for Motor 4. This pin can be assigned to either timer TIM4_CH4 or TIM11_CH1, neither of which have possible DMA assignments.
> 
> Additionally, Motor 2 is using pin B07, which can only be assigned to TIM4_CH2. This timer DMA assignment conflicts with SPI2_RX, which is further compounded by the fact you have two peripherals sharing that bus.
> 
> You must re-design the F405-based FCs to use sensible pins for the motor outputs so that each pin has both timer and DMA assignments available, and so that those assignments do not conflict with the timer/DMA assignments needed for other peripherals (e.g. gyro, OSD, flash, ADC, LED_STRIP). While you're at it, make sure each peripheral has it's own SPI bus.

Our FC Config file refers to the SpeedyBeeFC Config file in Betaflight, which should be no problem

**Comment by ot0tot** [comment]
> > > > > No DMA assignment is possible for Motor 4 on the F405 targets, which means the FC must use bitbanged DShot and the gyro on SPI 1 cannot use DMA. This seems like a major oversight and will result in poor performance. IMO these targets need to be redesigned and should not be approved as-is.
> > > > 
> > > > 
> > > > How should this be distributed?
> > > 
> > > 
> > > It has been modified, and you need to check whether it matches or not.
> > 
> > 
> > Your change is invalid. Re-read my comment.
> > You are using pin B09 for Motor 4. This pin can be assigned to either timer TIM4_CH4 or TIM11_CH1, neither of which have possible DMA assignments.
> > Additionally, Motor 2 is using pin B07, which can only be assigned to TIM4_CH2. This timer DMA assignment conflicts with SPI2_RX, which is further compounded by the fact you have two peripherals sharing that bus.
> > You must re-design the F405-based FCs to use sensible pins for the motor outputs so that each pin has both timer and DMA assignments available, and so that those assignments do not conflict with the timer/DMA assignments needed for other peripherals (e.g. gyro, OSD, flash, ADC, LED_STRIP). While you're at it, make sure each peripheral has it's own SPI bus.
> 
> Our FC Config file refers to the SpeedyBeeFC Config file in Betaflight, which should be no problem

The SPEEDYBEEF405V3 is the worst designed FC of the modern era and should only be used as an example of what NOT to do. 

Unfortunately, the FC designs here are actually worse because they use the faster ICM42688p gyro (8 kHz) vs the BMI270 (3.2kHz) used by the SpeedyBee FC. Thus you absolutely need DMA for the gyro, which is not possible due to your selected motor outputs.

**Comment by sugaarK** [comment]
> The SPEEDYBEEF405V3 is the worst designed FC of the modern era and should only be used as an example of what NOT to do.

Totally agree this FC is popular because of the war and it’s price. The fact it keeps been copied is mind boggling

**Pull request closed without merging**